### PR TITLE
feat(http-logger): allow verifying ssl cert

### PR DESCRIPTION
**Pull request by zhendongcmss**
### What this PR does / why we need it:

Fix: http-logger always use `false` on function `httpc:ssl_handshake`

### Pre-submission checklist:

* [ ] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: apisix/plugins/http-logger.lua [hunk 1]
```diff
@@ -45,7 +45,8 @@ local schema = {
             }
         },
         concat_method = {type = "string", default = "json",
-                         enum = {"json", "new_line"}}
+                         enum = {"json", "new_line"}},
+        ssl_verify = {type = "boolean", default = true},
```
Comment (spacewander) [inline apisix/plugins/http-logger.lua:49]: For compatibility, we have to use false by default.

Comment (zhendongcmss) [reply apisix/plugins/http-logger.lua:49]: ok

**Review by starsz — COMMENTED** [review]
Can you add a test case to cover if ssl.verify is true but ssl verify failed?

**Review by spacewander — APPROVED** [review]

**Comment by spacewander** [comment]
> Can you add a test case to cover if ssl.verify is true but ssl verify failed?

What's your idea about how to test it? It seems a bit tough to me.

**Review by tokers — DISMISSED** [review]

**Review by shuaijinchao — APPROVED** [review]

**Comment by starsz** [comment]
> > Can you add a test case to cover if ssl.verify is true but ssl verify failed?
> 
> @starsz What's your idea about how to test it? It seems a bit tough to me.

I think we can create a test case like `TEST 10: set correct https endpoint` and set the `ssl_verify` as true.

Then we can check the error log containing `failed to perform SSL with host`.

**Comment by spacewander** [comment]
@zhendongcmss 
Could you add the test according to the advice of @starsz before we can merge it?

**Review by tzssangglass — APPROVED** [review]

**Comment by zhendongcmss** [comment]
> @zhendongcmss Could you add the test according to the advice of @starsz before we can merge it?

Let me try

**Comment by zhendongcmss** [comment]
> > @zhendongcmss Could you add the test according to the advice of @starsz before we can merge it?
> 
> Let me try

The test cases always fail. There is a error log. can you have a look. @spacewander 
`2022/02/17 13:39:23 [error] 4278#4278: *2803266 [lua] batch-processor.lua:73: Batch Processor[http-logger] failed to process entries: failed to perform SSL with host[100.76.7.28] port[9999] handshake failed, context: ngx.timer, client: 127.0.0.1, server: 0.0.0.0:9080`

https://github.com/apache/apisix/runs/5226722265?check_suite_focus=true#step:14:875

**Comment by zhendongcmss** [comment]
Test case 9 already test the ssl handshake failed.

**Comment by starsz** [comment]
> Test case 9 already test the ssl handshake failed.

Hi, @zhendongcmss.They are different. Test case 9 test port 8888 that doesn't support https.
The case you need to write is added for ssl_verify failed.

File: t/plugin/http-logger.t [hunk 2]
```diff
@@ -820,3 +825,62 @@ failed to validate the 'include_resp_body_expr' expression: invalid operator '<>
 done
 --- no_error_log
 [error]
+
+
+
+=== TEST 19: ssl_verify default is false for comppatibaility
```
[the remark this answers is not in the document]
Comment (zhendongcmss) [reply t/plugin/http-logger.t:831]: ok

File: t/plugin/http-logger.t [hunk 3]
```diff
@@ -359,8 +362,9 @@ failed to perform SSL with host[127.0.0.1] port[8888] handshake failed
                                 "batch_max_size": 1,
                                 "max_retry_count": 1,
                                 "retry_delay": 2,
-                                "buffer_duration": 2,
-                                "inactive_timeout": 2
+                                "buffer_duration": 1,
+                                "inactive_timeout": 1,
+                                "ssl_verify": true
```
[the remark this answers is not in the document]
Comment (zhendongcmss) [reply t/plugin/http-logger.t]: ok

**Comment by starsz** [comment]
Hi @zhendongcmss.I think you still misunderstand my point.

> Could you use TEST 10, set "ssl_verify": true, and expect it to fail in the --- error_log?
As @spacewander and I suggested. You can add the follow test:

```
=== TEST 19: set correct https endpoint and ssl verify true
--- config
    location /t {
        content_by_lua_block {
            local t = require("lib.test_admin").test
            local code, body = t('/apisix/admin/routes/1',
                 ngx.HTTP_PUT,
                 [[{
                        "plugins": {
                            "http-logger": {
                                "uri": "https://127.0.0.1:9999/hello-world-http",
                                "batch_max_size": 1,
                                "max_retry_count": 1,
                                "retry_delay": 2,
                                "buffer_duration": 2,
                                "inactive_timeout": 2,
                                "ssl_verify": true
                            }
                        },
                        "upstream": {
                            "nodes": {
                                "127.0.0.1:1982": 1
                            },
                            "type": "roundrobin"
                        },
                        "uri": "/hello1"
                }]],
                [[{
                    "node": {
                        "value": {
                            "plugins": {
                                "http-logger": {
                                    "uri": "https://127.0.0.1:9999/hello-world-http",
                                    "batch_max_size": 1,
                                    "max_retry_count": 1,
                                    "retry_delay": 2,
                                    "buffer_duration": 2,
                                    "inactive_timeout": 2,
                                    "ssl_verify": true
                                }
                            },
                            "upstream": {
                                "nodes": {
                                    "127.0.0.1:1982": 1
                                },
                                "type": "roundrobin"
                            },
                            "uri": "/hello1"
                        },
                        "key": "/apisix/routes/1"
                   },
                    "action": "set"
                }]]
                )
            if code >= 300 then
                ngx.status = code
            end
            ngx.say(body)
        }
    }
--- request
GET /t
--- response_body
passed
--- no_error_log
[error]


=== TEST 20: access correct https endpoint but ssl verify failed
--- request
GET /hello1
--- error_log
self signed certificate in certificate chain
--- wait: 1.5
```

**Review by spacewander — APPROVED** [review]

**Review by lilien1010 — APPROVED** [review]
LGTM

**Pull request merged**